packages/inter-protocol/src/proposals/startPSM.js

```diff
@@ -18,24 +20,115 @@ import {
 } from './committee-proposal.js';
 
 /** @typedef {import('@agoric/vats/src/core/lib-boot.js').BootstrapManifest} BootstrapManifest */
+/** @typedef {import('../psm/psm.js').MetricsNotification} MetricsNotification */
+/** @typedef {import('./econ-behaviors.js').EconomyBootstrapPowers} EconomyBootstrapPowers */
 
 const BASIS_POINTS = 10000n;
-const { details: X } = assert;
+const { details: X, Fail } = assert;
 
 export { inviteCommitteeMembers, startEconCharter, inviteToEconCharter };
 
 /**
- * @param {EconomyBootstrapPowers & WellKnownSpaces} powers
+ * Decode vstorage value to CapData
+ * XXX already written somewhere?
+ *
+ * @param {unknown} value
+ */
+const decodeToCapData = value => {
+  assert.typeof(value, 'string');
+
+  // { blockHeight: 123, values: [ ... ] }
+  const item = JSON.parse(value); // or throw
+  assert.typeof(item, 'object');
+  assert(item);
+  const { values } = item;
+  assert(Array.isArray(values));
+
+  assert.equal(values.length, 1);
+  // { body: "...", slots: [ ... ] }
+  const data = JSON.parse(values[0]);
+  assert.typeof(data, 'object');
+  assert(data);
+  assert.typeof(data.body, 'string');
+  assert(Array.isArray(data.slots));
+
+  /** @type {import('@endo/marshal').CapData<string>} */
+  // @ts-expect-error cast
+  const capData = data;
+  return capData;
+};
+
+/**
+ * Provide access to object graphs serialized in vstorage.
+ *
+ * @param {Array<[string, string]>} entries
+ * @param {(slot: string, iface?: string) => any} [slotToVal]
+ */
+export const makeHistoryReviver = (entries, slotToVal = undefined) => {
+  const board = boardSlottingMarshaller(slotToVal);
+  const vsMap = makeMap(entries);
+
+  const getItem = key => {
+    const raw = vsMap.get(key) || Fail`no ${key}`;
+    const capData = decodeToCapData(raw);
+    return harden(board.fromCapData(capData));
+  };
+  const children = prefix => [
```

Comment:
Is this missing validation that `prefix` ends with "." to avoid e.g. `children("foo")` not filtering out a path like "foolish.donkey"?